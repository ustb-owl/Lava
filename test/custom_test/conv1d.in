1000
383 886 777 915 793 335 386 492 649 421 362 27 690 59 763 926 540 426 172 736 211 368 567 429 782 530 862 123 67 135 929 802 22 58 69 167 393 456 11 42 229 373 421 919 784 537 198 324 315 370 413 526 91 980 956 873 862 170 996 281 305 925 84 327 336 505 846 729 313 857 124 895 582 545 814 367 434 364 43 750 87 808 276 178 788 584 403 651 754 399 932 60 676 368 739 12 226 586 94 539 795 570 434 378 467 601 97 902 317 492 652 756 301 280 286 441 865 689 444 619 440 729 31 117 97 771 481 675 709 927 567 856 497 353 586 965 306 683 219 624 528 871 732 829 503 19 270 368 708 715 340 149 796 723 618 245 846 451 921 555 379 488 764 228 841 350 193 500 34 764 124 914 987 856 743 491 227 365 859 936 432 551 437 228 275 407 474 121 858 395 29 237 235 793 818 428 143 11 928 529 776 404 443 763 613 538 606 840 904 818 128 688 369 917 917 996 324 743 470 183 490 499 772 725 644 590 505 139 954 786 669 82 542 464 197 507 355 804 348 611 622 828 299 343 746 568 340 422 311 810 605 801 661 730 878 305 320 736 444 626 522 465 708 416 282 258 924 637 62 624 600 36 452 899 379 550 468 71 973 131 881 930 933 894 660 163 199 981 899 996 959 773 813 668 190 95 926 466 84 340 90 684 376 542 936 107 445 756 179 418 887 412 348 172 659 9 336 210 342 587 206 301 713 372 321 255 819 599 721 904 939 811 940 667 705 228 127 150 984 658 920 224 422 269 396 81 630 84 292 972 672 850 625 385 222 299 640 42 898 713 298 190 524 590 209 581 819 336 732 155 994 4 379 769 273 776 850 255 860 142 579 884 993 205 621 567 504 613 961 754 326 259 944 202 202 506 784 21 842 868 528 189 872 908 958 498 36 808 753 248 303 333 133 648 890 754 567 746 368 529 500 46 788 797 249 990 303 33 363 497 253 892 686 125 152 996 975 188 157 729 436 460 414 921 460 304 28 27 50 748 556 902 794 697 699 43 39 2 428 403 500 681 647 538 159 151 535 134 339 692 215 127 504 629 49 964 285 429 343 335 177 900 238 971 949 289 367 988 292 795 743 144 829 390 682 340 541 569 826 232 261 42 360 117 23 761 81 309 190 425 996 367 677 234 690 626 524 57 614 168 205 358 312 386 100 346 726 994 916 552 578 529 946 290 647 970 51 80 631 593 857 627 312 886 214 355 512 90 412 479 610 969 189 274 355 641 620 433 987 888 338 566 770 284 856 417 606 260 849 237 205 59 217 518 945 783 873 458 873 637 289 483 607 478 757 314 471 729 100 459 618 438 25 388 74 233 157 681 493 358 270 699 417 839 569 363 622 794 173 847 431 462 682 390 292 791 57 115 521 157 574 491 947 951 231 21 537 740 54 30 98 325 81 516 516 2 231 139 796 404 338 580 218 21 970 862 812 379 977 685 536 904 176 483 207 759 857 744 499 911 127 950 236 560 818 105 563 49 244 711 805 934 291 375 955 614 589 768 993 918 805 882 822 982 717 30 93 574 126 593 486 253 543 74 814 713 179 377 762 775 88 919 710 732 294 17 346 235 137 691 153 943 573 328 925 291 710 18 217 836 963 55 90 858 130 904 571 661 633 685 789 73 604 851 805 250 868 503 485 6 195 639 949 120 967 226 763 677 596 981 865 560 36 955 770 518 211 342 532 196 379 321 270 984 172 427 234 40 283 72 398 830 63 347 950 30 573 714 59 522 47 924 82 435 232 204 954 443 898 486 640 278 159 262 262 683 41 848 723 324 272 122 154 335 821 457 365 747 171 776 269 218 701 703 653 933 907 959 728 806 797 720 84 308 334 698 991 376 898 715 52 171 189 559 506 10 16 224 109 539 0 378 109 53 81 114 338 989 426 67 147 223 787 231 532 122 281 875 850 179 590 254 350 131 813 857 494 181 81 603 720 433 982 181 487 415 296 825 404 722 892 551 297 32 134 181 506 415 57 708 595 999 962 297 483 776 154 977 309 587 932 382 21 266 563 860 682 211 685 86 285 930 990 583 314 476 116 820 892 525 528 839 525 490 136 360 618 643 337 928 582 621 310 955 888 225 815 570 437 853 8 722 783 350 657 97 827 126 269 71 651 149 910 528 639 398 888 610 393 577 890 976 199 552 931 87 777 99 657 566 952 17 641 735 368 298 184 195 776 805 266 428 954 528 308 593 278 197 555 672 774 445 0 325 997 283 412 127 382 421 
693 334 439 334 421 159 985 957 354 761 762 972 541 716 852 850 662 482 399 217 154 173 15 506 851 364 790 263 491 172 37 537 859 828 871 280 987 856 590 341 970 352 665 511 69 517 361 83 351 112 300 506 638 667 364 489 32 154 104 875 679 141 412 538 969 636 170 956 844 760 649 814 465 314 326 886 183 39 969 535 152 621 393 790 289 109 631 673 264 735 548 295 877 313 833 198 949 355 155 793 468 156 960 933 823 286 171 358 677 140 245 181 761 990 323 50 100 954 75 364 42 624 659 919 289 844 469 238 551 976 383 19 133 343 304 956 981 475 666 11 967 912 192 729 902 868 131 2 174 207 718 216 183 377 487 472 573 957 62 125 933 797 496 418 141 153 726 474 980 393 485 948 305 30 29 559 898 160 562 424 719 280 641 902 10 480 726 583 789 140 708 723 938 557 493 431 710 220 905 690 613 391 638 270 421 667 829 671 180 743 95 899 24 88 154 386 569 232 969 710 373 30 433 663 587 279 94 649 499 351 339 464 742 330 86 515 349 915 186 881 11 634 133 387 722 287 773 643 519 742 354 244 124 139 259 63 418 353 712 269 705 404 733 799 734 819 315 435 87 853 669 450 487 802 837 562 89 610 205 960 704 911 557 829 403 816 892 821 522 605 443 579 361 528 378 447 700 45 882 787 899 551 589 386 353 426 948 794 36 506 107 92 417 664 921 820 480 166 994 354 123 437 933 484 317 312 931 17 709 165 156 608 69 745 995 422 171 295 569 559 801 676 652 571 340 925 743 172 91 89 527 566 878 812 50 196 124 333 213 186 499 370 794 568 115 141 342 639 437 263 198 590 939 202 513 631 128 257 804 571 346 683 138 225 495 540 421 972 226 634 158 725 356 952 645 472 446 339 111 883 603 661 825 542 216 339 174 344 596 330 267 294 13 757 519 860 650 292 832 876 279 990 953 635 295 598 107 741 937 570 976 540 584 801 83 800 492 609 496 440 939 763 735 304 873 606 164 523 251 349 751 530 339 704 165 986 302 625 79 591 547 407 484 131 561 919 931 53 528 427 494 819 543 581 123 768 187 639 643 438 988 394 320 680 98 486 18 752 463 98 695 10 505 179 142 66 98 425 472 978 853 966 797 748 547 272 516 86 912 159 877 900 553 197 932 3 35 951 107 498 49 154 861 906 334 3 325 784 428 797 763 633 115 560 381 14 185 897 100 97 408 329 349 313 879 634 316 914 585 775 765 986 930 626 892 616 629 569 752 409 366 515 395 833 428 776 847 613 26 300 62 786 629 763 100 508 397 416 775 982 544 540 320 826 518 565 794 499 134 546 908 853 62 303 686 842 432 534 807 810 834 869 596 815 984 696 324 382 465 451 716 361 343 37 187 861 602 981 360 88 879 620 941 293 924 628 135 708 162 942 870 996 163 466 811 500 515 487 234 332 290 950 693 633 339 880 494 293 213 854 382 444 475 323 738 751 951 873 811 465 168 681 813 683 499 977 535 14 464 769 346 107 72 391 740 411 623 587 57 836 793 439 281 620 114 371 371 418 596 534 883 764 567 697 800 67 674 335 433 490 457 132 949 529 523 690 292 147 629 349 335 422 140 968 43 255 339 766 25 936 653 260 52 220 957 204 287 983 540 721 826 997 205 127 878 728 169 170 227 798 520 563 221 660 883 616 267 223 382 292 511 35 553 563 608 862 768 895 198 660 968 376 9 173 503 887 254 673 57 481 823 929 396 44 942 280 12 209 855 747 854 366 134 759 281 742 973 401 638 171 413 958 899 422 484 755 661 90 780 71 923 603 0 320 0 942 952 12 504 807 759 358 525 894 117 158 636 90 560 626 614 973 937 865 748 421 620 409 863 400 832 786 4 833 458 356 127 410 368 631 569 128 341 446 374 458 605 10 901 165 989 867 490 926 732 238 699 705 0 562 457 832 348 461 17 807 169 497 569 538 128 139 18 470 585 392 280 542 754 533 707 743 400 550 21 485 788 720 190 140 634 647 325 983 461 694 142 630 191 63 520 320 554 538 142 492 930 422 34 685 308 94 780 708 644 802 545 784 522 87 925 157 735 602 492 548 296 986 530 840 49 51 512 956 589 654 448 872 428 482 557 88 576 337 797 220 139 694 5 14 782 282 523 869 236 15 417 884 353 299 724 754 350 236 710 292 242 158 164 23 641 721 111 569 410 260 790 902 955 147 916 89 781 439 958 17 806 375 901 511 674 978 265 377 566 976 669 161 134 833 
