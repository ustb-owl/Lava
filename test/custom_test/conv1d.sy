const int N = 1024;
void conv1d(int n, int A[], int B[], int C[]) {
    int i = 0;

    i = 0;
    while (i < n) {
        int j = 0;
        C[i] = 0;
        while (j <= i) {
            C[i] = C[i] + A[j] * B[i-j];
            j = j + 1;
        }
        i = i + 1;
    }
}

int A[N];
int B[N];
int C[N];

int main() {
    int n = getint();
    int i, j;

    i = 0;
    while (i < n){
        A[i] = getint();
        i = i + 1;
    }

    i = 0;
    while (i < n){
        B[i] = getint();
        i = i + 1;
    }

    starttime();
    i = 0;
    while (i < 1000) {
        conv1d(n, A, B, C);
        conv1d(n, B, C, A);
        i = i + 1;
    }
    int ans = 0;
    i = 0;
    while (i < n){
        ans = ans + A[i];
        i = i + 1;
    }
    stoptime();
    putint(ans);
    putch(10);
    return 0;
}