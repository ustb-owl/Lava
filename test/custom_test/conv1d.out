841117053
0
